steps/index.lab.md

```diff
@@ -215,123 +218,113 @@ Next we'll add some basic Firebase Authentication code to the app to detect user
+    val user = auth.currentUser
+    return user?.photoUrl?.toString()
 }
 
-private String getUserName() {
-    FirebaseUser user = mFirebaseAuth.getCurrentUser();
-    if (user != null) {
-        return user.getDisplayName();
-    }
-
-    return ANONYMOUS;
+private fun getUserName(): String? {
+    val user = auth.currentUser
+    return if (user != null) {
+        user.displayName
+    } else ANONYMOUS
 }
 ```
 
 Then implement the `signOut()` method to handle the sign out button:
 
-**MainActivity.java**
+**MainActivity.kt**
 
 ```
-private void signOut() {
-    mFirebaseAuth.signOut();
-    mSignInClient.signOut();
-    startActivity(new Intent(this, SignInActivity.class));
-    finish();
+private fun signOut() {
+    auth.signOut()
+    signInClient.signOut()
+    startActivity(Intent(this, SignInActivity::class.java))
+    finish()
 }
 ```
 
 Now we have all of the logic in place to send the user to the sign-in screen when necessary. Next we need to implement the sign-in screen to properly authenticate users.
 
 #### Implement the Sign-In screen
 
-Open the file `SignInActivity.java`.  Here a simple Sign-In button is used to initiate authentication. In this step you will implement the logic to Sign-In with Google, and then use that Google account to authenticate with Firebase.
+Open the file `SignInActivity.kt`.  Here a simple Sign-In button is used to initiate authentication. In this step you will implement the logic to Sign-In with Google, and then use that Google account to authenticate with Firebase.
 
 Add an Auth instance variable in the `SignInActivity` class under the `// Firebase instance variables` comment:
 
-**SignInActivity.java**
+**SignInActivity.kt**
 
 ```
 // Firebase instance variables
-private FirebaseAuth mFirebaseAuth;
+private lateinit var auth: FirebaseAuth
 ```
 
 Then, edit the `onCreate()` method to initialize Firebase in the same way you did in `MainActivity`:
 
-**SignInActivity.java**
+**SignInActivity.kt**
 
 ```
 // Initialize FirebaseAuth
-mFirebaseAuth = FirebaseAuth.getInstance();
+auth = Firebase.auth
 ```
 
 Next, initiate signing in with Google. Update `SignInActivity` `signIn()` method to look like this:
 
-**SignInActivity.java**
+**SignInActivity.kt**
 
 ```
-private void signIn() {
-    Intent signInIntent = mSignInClient.getSignInIntent();
-    startActivityForResult(signInIntent, RC_SIGN_IN);
+private fun signIn() {
+    val signInIntent = signInClient.signInIntent
+    startActivityForResult(signInIntent, RC_SIGN_IN)
 }
 ```
 
 Next, implement the `onActivityResult` method to `SignInActivity` to handle the sign in result. If the result of the Google Sign-In was successful, use the account to authenticate with Firebase:
 
-**SignInActivity.java**
+**SignInActivity.kt**
 
 ```
-@Override
-public void onActivityResult(int requestCode, int resultCode, Intent data) {
-    super.onActivityResult(requestCode, resultCode, data);
+super.onActivityResult(requestCode, resultCode, data)
```

Comment:
nit: I think showing the function signature is helpful here, rather than just the function body.